### PR TITLE
Refactor and test requiring user-provided module

### PR DESCRIPTION
**Pull request by honzajavorek**
#### :memo: Related issues and Pull Requests

Factored out from https://github.com/apiaryio/dredd/pull/1289/

#### :white_check_mark: What didn't I forget?

- [ ] To write docs
- [x] To write tests
- [x] To put [Conventional Changelog](https://dredd.org/en/latest/internals.html#sem-rel) prefixes in front of all my commits and run `npm run lint`

### CODE REVIEW

File: lib/resolveModule.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+const fs = require('fs');
+const path = require('path');
+
+
+module.exports = function resolveModule(workingDirectory, moduleName) {
+  const absolutePath = path.resolve(workingDirectory, moduleName);
+  return fs.existsSync(absolutePath) || fs.existsSync(`${absolutePath}.js`)
+    ? path.resolve(absolutePath)
```
Comment (artem-zakharchenko) [inline lib/resolveModule.js]: Why should we resolve already resolved path once more?

File: test/unit/resolveModule-test.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    assert.equal(
+      resolveModule(workingDirectory, 'requiredModule.js'),
+      path.join(workingDirectory, 'requiredModule.js')
+    );
+  });
+
+  it('resolves an installed module name', () => {
+    assert.equal(
+      resolveModule(workingDirectory, 'coffeescript/register'),
+      'coffeescript/register'
```
Comment (artem-zakharchenko) [inline test/unit/resolveModule-test.js]: A minor thing, but maybe it's more future-proof to mock some generic module (i.e. `mocha`), the one that is unlikely to be removed as the library's dependency. I assume `coffeescript` is a legacy, so it's going to be removed in the future. It's nice not to hook up to legacy things in the tests.

Comment (honzajavorek) [reply test/unit/resolveModule-test.js]: CoffeeScript is definitely a legacy as Dredd's direct dependency, but it is an existing JS transpiler (it's still developed and some people still use it). The `--require` option was introduced to mainly support transpilers, similar to `mocha --compilers js:babel-core/register`.

Any transpiler would do to verify the feature works, but `coffeescript` was already in dev dependencies from the past. Its objective advantage is it has zero dependencies itself, so using it to test the feature is very "cheap" (compared to e.g. `babel-core`, which has ~14 dependencies right now, and might require additional settings or packages to actually transpile something).

Also... it is a tiny little tribute to Dredd's origins... okay, with this one I admit I'm just making up things now 😄

File: lib/resolveModule.js (hunk 1 above)
Comment (honzajavorek) [reply lib/resolveModule.js]: That's a good question. Looks like a leftover from some furious changes in code. I'll play with the code for a while and get back to address this comment.

Comment (honzajavorek) [reply lib/resolveModule.js]: Okay, verified it has no particular reason. Addressed in da645b2

**Pull request merged**